packages/actor-bindings-aggregator-factory-average/lib/AverageAggregator.ts

```diff
@@ -0,0 +1,52 @@
+import type { ExpressionEvaluator } from '@comunica/expression-evaluator';
+import { AggregateEvaluator } from '@comunica/expression-evaluator';
+import * as E from '@comunica/expression-evaluator/lib/expressions';
+import { regularFunctions } from '@comunica/expression-evaluator/lib/functions';
+import { integer } from '@comunica/expression-evaluator/lib/functions/Helpers';
+import * as C from '@comunica/expression-evaluator/lib/util/Consts';
+import type { IActionContext, IExpressionEvaluatorFactory, IBindingAggregator } from '@comunica/types';
+import type * as RDF from '@rdfjs/types';
+import type { Algebra } from 'sparqlalgebrajs';
+
+interface IAverageState {
+  sum: E.NumericLiteral;
+  count: number;
+}
+
+export class AverageAggregator extends AggregateEvaluator implements IBindingAggregator {
+  // This will eventually be a mediator call.
+  private readonly summer = regularFunctions[C.RegularOperator.ADDITION];
+  private readonly divider = regularFunctions[C.RegularOperator.DIVISION];
```

Comment:
Instead of accessing these directly, I think it should be nicer to ask the function to either an EE instance, or ask for a function instance of the EE-Factory. (These things should be considered when functions have been extracted (and they will be considered because extraction will kill this code))

Reply:
(this can be ignored, I just wanted to clarify this)